src/gt4py/next/program_processors/runners/dace_fieldview/gtir_sdfg.py

```diff
@@ -413,73 +448,35 @@ def _add_sdfg_params(
             # the connectivity tables that are not used. The remaining unused transient arrays
             # are removed by the dace simplify pass.
             self._add_storage(
-                sdfg, symbolic_arguments, dace_utils.connectivity_identifier(offset), gt_type
+                symbolic_arguments, dace_utils.connectivity_identifier(offset), gt_type
             )
 
         # the list of all sdfg arguments (aka non-transient arrays) which include tuple-element fields
         return [arg_name for arg_name, _ in sdfg_args]
 
-    def visit_Program(self, node: gtir.Program) -> dace.SDFG:
-        """Translates `ir.Program` to `dace.SDFG`.
-
-        First, it will allocate field and scalar storage for global data. The storage
-        represents global data, available everywhere in the SDFG, either containing
-        external data (aka non-transient data) or temporary data (aka transient data).
-        The temporary data is global, therefore available everywhere in the SDFG
-        but not outside. Then, all statements are translated, one after the other.
-        """
-        if node.function_definitions:
-            raise NotImplementedError("Functions expected to be inlined as lambda calls.")
+    def visit_Program(self, node: gtir.Program) -> None:
+        """Translates the body of `ir.Program` inside the SDFG that belongs to this builder."""
 
         # Since program field arguments are passed to the SDFG as full-shape arrays,
         # there is no offset that needs to be compensated.
         assert len(self.field_offsets) == 0
 
-        sdfg = dace.SDFG(node.id)
-        sdfg.debuginfo = dace_utils.debug_info(node, default=sdfg.debuginfo)
-
-        # DaCe requires C-compatible strings for the names of data containers,
-        # such as arrays and scalars. GT4Py uses a unicode symbols ('ᐞ') as name
-        # separator in the SSA pass, which generates invalid symbols for DaCe.
-        # Here we find new names for invalid symbols present in the IR.
-        node = dace_gtir_utils.replace_invalid_symbols(sdfg, node)
-
-        # start block of the stateful graph
-        entry_state = sdfg.add_state("program_entry", is_start_block=True)
+        # create start block of the stateful graph
+        assert len(self.sdfg.nodes()) == 0
```

Comment:
I would make this a hard test not just an assert, or at least add a description.

Reply:
Yes, I had similar thoughts.